### PR TITLE
Stencilize/copy corners

### PR DESCRIPTION
**Pull request by rheacangeo**
## Purpose

Python code is killing our performance. The copy_corners method has patterns we can't generalize a stencil from without a gt4py feature. For now, hard code in 3 halo lines and copy the corners using regions. We will probably need to revisit this, but this PR takes a step forward, and so other stencils can run copy_corners_x or copy_corners_y as functions rather than breaking from the stencil to do so. 

## Code changes:
- gtscript functions copy_corners_x and copy_corners_y replace the python equivalents
- the copy_corners method in fv3core/utils/corners.py now calls a stencil, calling either copy_corners_x or copy_corners_y. 

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [x] Unit tests are added or updated for non-stencil code changes

### CODE REVIEW

**Comment by rheacangeo** [comment]
@jdahm feel free to merge this into the del2cubed branch, or if you approve I can merge it into master
@oelbert you can use this for delnflux

File: fv3core/utils/corners.py [hunk 1]
```diff
@@ -220,52 +221,116 @@ def definition(q: FloatField):
...
-def copy_sw_corner(q, direction, grid, kslice):
-    for j in range(grid.js - grid.halo, grid.js):
-        for i in range(grid.is_ - grid.halo, grid.is_):
-            if direction == "x":
-                q[i, j, kslice] = q[j, grid.is_ - i + 2, kslice]
-            if direction == "y":
-                q[i, j, kslice] = q[grid.js - j + 2, i, kslice]
+@gtscript.function
+def copy_corners_x(q):
+    from __externals__ import i_end, i_start, j_end, j_start
```
Comment (twicki) [inline fv3core/utils/corners.py:226]: why end before start?

File: fv3core/utils/corners.py [hunk 2]
```diff
@@ -220,52 +221,116 @@ def definition(q: FloatField):
...
-                q[i, j, kslice] = q[j + 2 - grid.ie, grid.je + 1 - i + 2, kslice]
+@gtstencil
+def copy_corners_y_stencil(q: FloatField):
+    with computation(PARALLEL), interval(...):
+        q = copy_corners_y(q)
 
 
-# can't actually be a stencil because offsets are variable
+# TODO: may want to put this in the test code if no longer used
+# directly in the model
```
Comment (twicki) [inline fv3core/utils/corners.py:323]: seems like with your refactor of the test, this is not used at all anymore?

**Review by twicki — CHANGES_REQUESTED** [review]
one minor comment. Thanks a lot for this work! Looks very tedious

File: fv3core/utils/corners.py (hunk 1 above)
Comment (rheacangeo) [reply fv3core/utils/corners.py:226]: good catch! no idea! I'll change it

Comment (rheacangeo) [reply fv3core/utils/corners.py:226]: Oh, looks like other gtscript functions in this file use that order, I probably copied it

File: fv3core/utils/corners.py (hunk 2 above)
Comment (rheacangeo) [reply fv3core/utils/corners.py:323]: yes! I meant to delete this

**Review by twicki — APPROVED** [review]

**Comment by rheacangeo** [comment]
launch jenkins

**Comment by rheacangeo** [comment]
launch jenkins

**Pull request merged**